packages/merkle-distributor/scripts/2_PublishClaimsForWindow.ts

```diff
@@ -0,0 +1,126 @@
+
+const assert = require("assert");
+import { program } from "commander";
+import fs from "fs";
+
+const { getAbi } = require("@uma/core");
+const { getWeb3, MAX_UINT_VAL } = require("@uma/common");
+const web3 = getWeb3();
+const { toBN } = web3.utils;
+
+import IpfsHelper = require("../src/IpfsHelper");
+import CloudflareKVHelper = require("../src/CloudflareKVHelper");
+
+program
+  .option("-n --network")
+  .requiredOption("-i, --input <path>", "input JSON file location containing a recipients payout")
+  .requiredOption("-m, --merkleDistributorAddress <address>", "address of the merkle distributor contract")
+  .parse(process.argv);
+
+const options = program.opts();
+
+const claimsObject = JSON.parse(fs.readFileSync(options.input, { encoding: "utf8" }));
+
+// We can't easily do runtime verification of JSON file types in typescript. We could use a JSON scheme, but to keep
+// things simple for now we can just double check that some important keys are present within the JSON file.
+if (typeof claimsObject !== "object") throw new Error("Invalid JSON");
+const expectedKeys = ["chainId", "rewardToken", "windowIndex", "totalRewardsDistributed", "windowStart", "claims"];
+expectedKeys.forEach(expectedKey => {
+  if (!Object.keys(claimsObject).includes(expectedKey)) {
+    throw new Error("claims object missing expected key");
+  }
+});
+
+const merkleDistributor = new web3.eth.Contract(getAbi("MerkleDistributor"), options.merkleDistributorAddress);
+
+async function main() {
+  const account = (await web3.eth.getAccounts())[0];
+
+  // Do some basic sanity checks.
+  console.log("0. Running some basic sanity checks on the payout file 🔎");
+  assert(web3.utils.isAddress(options.merkleDistributorAddress), "Invalid merkleDistributorAddress");
+  assert((await merkleDistributor.methods.owner().call()) == account, "Unlocked account does not own the distributor");
+  assert((await web3.eth.net.getId()) == claimsObject.chainId, "The connected network not match your  JSON chainId");
+  assert(web3.utils.isAddress(claimsObject.rewardToken), "Invalid rewardToken" + claimsObject.rewardToken);
+  assert((await merkleDistributor.methods.lastSeededIndex().call()) == claimsObject.windowIndex, "Wrong windowIndex");
+
+  console.log("File passed the checks\n\n1. Adding claims file to IPFS 🛰");
+
+  const ipfsHash = await IpfsHelper.uploadFile(JSON.stringify(claimsObject.claims));
+
+  console.log("Claims file added to IPFS with hash:", ipfsHash, "\n\n2. Pinning claims file to IPFS 📌");
+
+  await IpfsHelper.pinHash(ipfsHash);
+
+  console.log("Claims file pinned to IPFS!\n\n3. Adding claims file to cloudflare KV 🗺");
+
+  await CloudflareKVHelper.addClaimsToKV(claimsObject.claims, claimsObject.chainId, claimsObject.windowIndex);
+
+  console.log("Claims file added to cloudflareKV!\n\n4. Updating the lookup claims indices 🔭");
+
+  await CloudflareKVHelper.updateChainWindowIndicesFromKV(
```

Comment:
If one of the following steps breaks, is there any clean up thats neccessary to do or is it safe to re-upload or pin IPFS files for example:

1. Upload file to IPFS
2. Pin file hash to IPFS
3. Upload claims to cloudflare
4. Update look claims indices on cloudflare

Reply:
If it breaks we can do the remaining steps manually pretty easily. As long as the merkle distribution(created in script 1) works as expected, we can do all the other steps on our own. This is the main reason I split the script.